### PR TITLE
Added cell calculation function

### PR DESCRIPTION
**Pull request by maximpristupa**

### CODE REVIEW

File: evaluate.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
+Array.prototype.clean = function() {
```
Comment (YanaYaroshevich) [inline evaluate.js]: in my opinion you do not need this one
use build-in array filter method

File: evaluate.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    for(let i = 0; i < this.length; i++) {
+        if(this[i] === "") {
```
Comment (YanaYaroshevich) [inline evaluate.js]: !this[i] is enough, no?

File: evaluate.js [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+            this.splice(i, 1);
+        }
+    }
+    return this;
+}
+
+function expressionToRpn(expression) {
+        let outputQueue = "";
+        let operatorStack = [];
+        let operators = {
```
Comment (YanaYaroshevich) [inline evaluate.js]: it is a const, which won't be change
store it outside the function

File: evaluate.js [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+                associativity: "Left"
+            },
+            "-": {
+                precedence: 2,
+                associativity: "Left"
+            }
+        }
+        expression = expression.replace(/\s+/g, "");
+        expression = expression.split(/([\+\-\*\/\^\(\)])/).clean();
+        for(let i = 0; i < expression.length; i++) {
```
Comment (YanaYaroshevich) [inline evaluate.js]: please use build-in functions
forEach here

File: evaluate.js [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+            let token = expression[i];
+            if(token == +token) {
```
Comment (YanaYaroshevich) [inline evaluate.js]: follow spacing after key words if (, for (, else {

Comment (YanaYaroshevich) [reply evaluate.js]: avoid using '==', it means you don't know the types
always use '===' and '!=='

File: evaluate.js [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+                outputQueue += token + " ";
+            } else if("^*/+-".indexOf(token) !== -1) {
+                let o1 = token;
+                let o2 = operatorStack[operatorStack.length - 1];
+                while("^*/+-".indexOf(o2) !== -1 && ((operators[o1].associativity === "Left" && operators[o1].precedence <= operators[o2].precedence) || (operators[o1].associativity === "Right" && operators[o1].precedence < operators[o2].precedence))) {
```
Comment (YanaYaroshevich) [inline evaluate.js]: looks complex, move to function with a proper name

File: evaluate.js [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+                    outputQueue += operatorStack.pop() + " ";
```
Comment (YanaYaroshevich) [inline evaluate.js]: use string templates

File: evaluate.js [hunk 8]
```diff
@@ -0,0 +1,85 @@
+Array.prototype.clean = function() {
+    for(let i = 0; i < this.length; i++) {
+        if(this[i] === "") {
+            this.splice(i, 1);
+        }
+    }
+    return this;
+}
+
+function expressionToRpn(expression) {
```
Comment (YanaYaroshevich) [inline evaluate.js]: split it to smaller logical blocks
it would be better, and more readable

File: evaluate.js [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+            outputQueue += operatorStack.pop() + " ";
+        }
+        return outputQueue.replace('^', '**');
+    }
+
+function calculateRpnExpression(str) {
+    s = []
+    str = str.split(' ')
+    for (var i = 0; i < str.length; i++) {
+        if (str[i] == +str[i]) s.push(str[i])
```
Comment (YanaYaroshevich) [inline evaluate.js]: always use {, }

File: evaluate.js [hunk 10]
```diff
@@ -0,0 +1,85 @@
...
+        else if (str[i] == '+') s.push(+s.splice(-2,1)[0] + +s.pop())
```
Comment (YanaYaroshevich) [inline evaluate.js]: no '=='

File: evaluate.js [hunk 11]
```diff
@@ -0,0 +1,85 @@
...
+        }
+        while(operatorStack.length > 0) {
+            outputQueue += operatorStack.pop() + " ";
+        }
+        return outputQueue.replace('^', '**');
+    }
+
+function calculateRpnExpression(str) {
+    s = []
+    str = str.split(' ')
```
Comment (YanaYaroshevich) [inline evaluate.js]: please don't
one variable for a type
if you do anything new, create a new variable

File: evaluate.js [hunk 12]
```diff
@@ -0,0 +1,85 @@
...
+    for(let i = 0; i < this.length; i++) {
+        if(this[i] === "") {
+            this.splice(i, 1);
+        }
+    }
+    return this;
+}
+
+function expressionToRpn(expression) {
+        let outputQueue = "";
```
Comment (YanaYaroshevich) [inline evaluate.js]: read about difference between let and const

File: evaluate.js [hunk 13]
```diff
@@ -0,0 +1,85 @@
...
+            }
+        }
+        while(operatorStack.length > 0) {
+            outputQueue += operatorStack.pop() + " ";
+        }
+        return outputQueue.replace('^', '**');
+    }
+
+function calculateRpnExpression(str) {
+    s = []
```
Comment (YanaYaroshevich) [inline evaluate.js]: const?

Comment (YanaYaroshevich) [reply evaluate.js]: and better names would be great to have

File: evaluate.js [hunk 14]
```diff
@@ -0,0 +1,85 @@
...
+    str = str.split(' ')
+    for (var i = 0; i < str.length; i++) {
+        if (str[i] == +str[i]) s.push(str[i])
+        else if (str[i] == '+') s.push(+s.splice(-2,1)[0] + +s.pop())
+        else if (str[i] == '-') s.push(+s.splice(-2,1)[0] - +s.pop())
+        else if (str[i] == '*') s.push(+s.splice(-2,1)[0] * +s.pop())
+        else if (str[i] == '/') s.push(+s.splice(-2,1)[0] / +s.pop())
```
Comment (YanaYaroshevich) [inline evaluate.js]: a lot of repeating code
and witch case seems better here

File: evaluate.js (hunk 1 above)
Comment (maximpristupa) [reply evaluate.js]: changed as recommended

File: evaluate.js (hunk 2 above)
Comment (maximpristupa) [reply evaluate.js]: changed as recommended

File: evaluate.js (hunk 3 above)
Comment (maximpristupa) [reply evaluate.js]: changed as recommended

File: evaluate.js (hunk 4 above)
Comment (maximpristupa) [reply evaluate.js]: changed as recommended

File: evaluate.js (hunk 8 above)
Comment (maximpristupa) [reply evaluate.js]: slightly cleaned the function

File: evaluate.js (hunk 14 above)
Comment (maximpristupa) [reply evaluate.js]: changed to switch case

**Pull request merged**